### PR TITLE
docs: update walkthrough guide

### LINKED ISSUE #1239: Update walkthrough documentation
## WHAT

Revisit those chapters of the `management-api-walkthrough` and updates where needed:

- Assets
- Policies accordingly with the spec available in [tractusx-profiles](https://github.com/eclipse-tractusx/tractusx-profiles/blob/main/cx/policy/specs/policy.definitions.md)
- Catalog
- Contract Definition
- Contract Negotiations
- Transfer process
- Contract agreements

### PR DESCRIPTION
**Pull request by wolf4ood**
## WHAT

updates the walkthrough mostly by adjusting payloads and adding 0.7.x relevant documentation

## WHY

documentation

Closes #1239

### CODE REVIEW

File: docs/usage/management-api-walkthrough/02_policies.md [hunk 1]
```diff
@@ -28,38 +28,33 @@ impacting policy definitions.
 Policies can be created in the EDC as follows:
 
 ```http
-POST /v3/assets HTTP/1.1
+POST /v3/policydefinitions HTTP/1.1
```
Comment (arnoweiss) [inline docs/usage/management-api-walkthrough/02_policies.md:31]: lol, good catch

File: docs/usage/management-api-walkthrough/04_catalog.md [hunk 2]
```diff
@@ -14,23 +14,19 @@ Content-Type: application/json
...
     "offset": 0,
     "limit": 50,
     "sortField": "https://purl.org/dc/terms/type",
     "sortOrder": "ASC",
     "filterExpression": [
       {
-        "@context": {
-          "edc": "https://w3id.org/edc/v0.0.1/ns/"
-        },
-        "@type": "edc:Criterion",
```
Comment (arnoweiss) [inline docs/usage/management-api-walkthrough/04_catalog.md:33]: Is this tested? In the past, `leftOperands` had to be given in expanded format (so it appears this was wrong in the docs before).

File: docs/usage/management-api-walkthrough/04_catalog.md [hunk 3]
```diff
@@ -42,6 +38,8 @@ Content-Type: application/json
 The request body is lean. Mandatory properties are:
 - `counterPartyAddress` (formerly `providerUrl`): This property points to the DSP-endpoint of the Data Provider's Control
   Plane. Usually this ends on `/api/v1/dsp`.
+- `counterPartyId`: must be the provider BPN. This in not a mandatory property for EDC but it's required for TractusX-EDC.
+  If omitted the catalog request will fail.
```
Comment (arnoweiss) [inline docs/usage/management-api-walkthrough/04_catalog.md:42]: just curious: why does tractusx-edc require this additional bit of info in 0.7.x?

File: docs/usage/management-api-walkthrough/04_catalog.md [hunk 4]
```diff
@@ -148,13 +167,12 @@ Consequently, if there may be more than one offer for the same Asset, requiring
 policies included.
 
 - The `@id` corresponds to the id of the Asset that can be negotiated for.
-- `dcat:Distribution` makes statements over which Data Planes an Asset's data can be retrieved. Currently, the EDC always
-  returns `HttpProxy` and `AmazonS3` even though not all EDC-deployments have both capabilities.
+- `dcat:Distribution` makes statements over which Data Planes an Asset's data can be retrieved. Currently, the TractusX-EDC supports
```
Comment (arnoweiss) [inline docs/usage/management-api-walkthrough/04_catalog.md:170]: Does this statement hold true only for the general capabilities of the Dataplane OR for also for this particular `dcat:Dataset`? With 0.6.x, I've witnessed it just returning `HttpData-PULL`, `AmazonS3-PUSH` and `AzureStorage-PUSH` independently of how the `Asset `was registered.

File: docs/usage/management-api-walkthrough/08_contractagreements.md [hunk 5]
```diff
@@ -15,41 +15,44 @@ A Contract Agreement looks like this:
...
   "@context": {
-    "dct": "https://purl.org/dc/terms/",
-    "tx": "https://w3id.org/tractusx/v0.0.1/ns/",
+    "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
     "edc": "https://w3id.org/edc/v0.0.1/ns/",
-    "dcat": "https://www.w3.org/ns/dcat/",
-    "odrl": "http://www.w3.org/ns/odrl/2/",
-    "dspace": "https://w3id.org/dspace/v0.8/"
+    "tx": "https://w3id.org/tractusx/v0.0.1/ns/",
+    "tx-auth": "https://w3id.org/tractusx/auth/",
```
Comment (arnoweiss) [inline docs/usage/management-api-walkthrough/08_contractagreements.md:53]: I haven't seen this namespace used anywhere - what does it manage?

**Review by arnoweiss — APPROVED** [review]
Thanks for this PR - clarifies and updates a lot. At some point, a separate PR could update the documentation of the `/edrs` API as that appears to have changed a lot too. There's some errors in unchanged sections too that don't have to be included for this PR to be merged. Got a couple of questions but nothing blocking.

File: docs/usage/management-api-walkthrough/08_contractagreements.md (hunk 5 above)
Comment (wolf4ood) [reply docs/usage/management-api-walkthrough/08_contractagreements.md:53]: This are automatically returned by EDC side in every response when we register custom namespaces. event if not used in this specific response

**Comment by wolf4ood** [comment]
@arnoweiss  the EDR api docs has already been modified in [this](https://github.com/eclipse-tractusx/tractusx-edc/pull/1235) PR.

**Review by jimmarino — APPROVED** [review]

File: docs/usage/management-api-walkthrough/04_catalog.md (hunk 2 above)
Comment (wolf4ood) [reply docs/usage/management-api-walkthrough/04_catalog.md:33]: do you mean the `edc:someProperty` ?

Comment (arnoweiss) [reply docs/usage/management-api-walkthrough/04_catalog.md:33]: yep, `https://w3id.org/edc/v0.0.1/ns/someProperty` I've seen work in the past, `edc:someProperty` (even with appropriate context present) not.

File: docs/usage/management-api-walkthrough/04_catalog.md (hunk 4 above)
Comment (wolf4ood) [reply docs/usage/management-api-walkthrough/04_catalog.md:170]: That's independent on how the Asset is configured. The asset only says where the is resides and in which protocol, but that is the source of a transfer and the consumer does not see that. 

`HttpData-PULL`,  `AmazonS3-PUSH` and `AzureStorage-PUSH` are the destination, or how the consumer can receive data.

Those can be mixed, for example if I configure an asset with `HttpData` that can be consumed:

- HttpData-PULL (Consumer pull the data) 
- AmazonS3-PUSH/AzureStorage-PUSH (Provider push the data)

File: docs/usage/management-api-walkthrough/04_catalog.md (hunk 2 above)
Comment (wolf4ood) [reply docs/usage/management-api-walkthrough/04_catalog.md:33]: Yes, it was already in the form of `edc:someProperty` but is not correct. Will fix it

**Pull request merged**